File: src/main/java/xyz/baddeveloper/lwsl/server/Controller.java
package xyz.baddeveloper.lwsl.server;

import xyz.baddeveloper.lwsl.packet.Packet;

import java.io.IOException;
import java.net.Socket;

public class Controller {

    private SocketServer socketServer;

    public Controller(SocketServer socketServer) {
        this.socketServer = socketServer;
    }

    public void sendPacketToAll(Packet packet){
        socketServer.getClients().forEach(socketHandler -> socketHandler.sendPacket(packet));
    }

    public void kickAll() {
        socketServer.getClients().forEach(this::kickClient);
    }

    public SocketHandler getClientByIP(String hostname){
        for(SocketHandler socketHandler : socketServer.getClients()){
            if(socketHandler.getSocket().getInetAddress().getHostName().equals(hostname))
                return socketHandler;
        }
        return null;
    }

    public void kickClient(Socket socket) {
        try {
            socket.close();
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    public void kickClient(SocketHandler client) {
        try {
            client.getSocket().close();
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: src/main/java/com/firenodes/lwsl/server/Controller.kt
package com.firenodes.lwsl.server

import com.firenodes.lwsl.packet.Packet
import java.io.IOException
import java.net.InetSocketAddress
import java.net.Socket
import java.util.function.Consumer


class Controller(private val socketServer: SocketServer) {
    fun sendPacketToAll(packet: Packet) {
        socketServer.clients.values.forEach(Consumer { socketHandler: SocketHandler? -> socketHandler!!.sendPacket(packet) })
    }

    fun kickAll() {
        socketServer.clients.values.forEach(Consumer { client: SocketHandler? -> this.kickClient(client) })
    }

    fun getClientByIP(hostname: String): SocketHandler? {
        for (socketHandler in socketServer.clients) {
            if ((socketHandler.value?.connection?.channel()?.remoteAddress() as InetSocketAddress).address.hostAddress == hostname) return socketHandler.value
        }
        return null
    }

    fun kickClient(socket: Socket) {
        try {
            socket.close()
        } catch (e: IOException) {
            e.printStackTrace()
        }
    }

    fun kickClient(client: SocketHandler?) {
        client?.connection?.dispose()
    }

}
